### PR TITLE
Build: discord v13 support, slashcommands support.

### PR DESCRIPTION
**Pull request by DhansAL**
# Description

This pull request resolves issues #143  #102
 - Added support to discord `v13.4.0`.
 - Restructured code, created specific event handlers,modules,etc.
 - Added support for slashCommands
 - Added support for @discordjs/rest package
 - Documented events and modules
 - Removed deprecated code.
 
Above mentioned points are the summary of the changes made to the source code.
This pull request makes considerable changes to the code structure and current code of the bot. This pull request may contain some bugs but I have tested all the commands and made respective error handlers in the code. feel free to mention any issues :)

## Check List (mark all applicable checkboxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots (if applicable)

 Original           | Updated
 :--------------------: |:--------------------:
 original screenshot | updated screenshot |

### CODE REVIEW

**Comment by DhansAL** [comment]
omg this must be having a lot of typos (T_T) I'll try to make as less as possible @Panquesito7

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Hey there! I just merged #142. Can you please resolve the merge conflicts [**here**](https://github.com/commclassroom/classroom-monitor-bot/pull/146/conflicts)? Let us know if you need any help! 😃
Here's a very detailed guide on how to fix merge conflicts, from GitHub Docs: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github

**Comment by DhansAL** [comment]
sure! will do this asap

**Comment by DhansAL** [comment]
i have removed all the conflicts, you can review the code now ^_^

**Comment by DhansAL** [comment]
Is anyone having any issues with this pull request? I'll be happy to help ^_^

**Review by SiddhantKhisty — APPROVED** [review]
Looks good to me

File: package.json [hunk 1]
```diff
@@ -1,60 +1,65 @@
...
+		"fs": "^0.0.1-security",
+		"memejs": "^2.0.1",
+		"winston": "^3.3.3"
+	},
+	"engines": {
+		"node": "16.x",
+		"npm": "7.x"
+	}
+}
+
```
Comment (Panquesito7) [inline package.json:65]:
```suggestion
```

**Review by Panquesito7 — APPROVED** [review]
Let's give it a go! 🚀 Thanks so much for your contribution! 😄👍

**Pull request merged**